libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/BigIcon.kt

```diff
@@ -0,0 +1,154 @@
+ */
+
+package io.element.android.libraries.designsystem.components
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.CatchingPokemon
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import io.element.android.compound.theme.ElementTheme
+import io.element.android.compound.tokens.generated.CompoundIcons
+import io.element.android.libraries.designsystem.preview.ElementPreview
+import io.element.android.libraries.designsystem.preview.PreviewsDayNight
+import io.element.android.libraries.designsystem.theme.bigIconDefaultBackgroundColor
+import io.element.android.libraries.designsystem.theme.components.Icon
+import io.element.android.libraries.ui.strings.CommonStrings
+
+/**
+ * Compound component that display a big icon centered in a rounded square.
+ */
+object BigIcon {
+    /**
+     * The style of the [BigIcon].
+     */
+    @Immutable
+    sealed interface Style {
+        /**
+         * The default style.
+         *
+         * @param vectorIcon the [ImageVector] to display
+         * @param contentDescription the content description of the icon, if any. It defaults to `null`
+         */
+        data class Default(val vectorIcon: ImageVector, val contentDescription: String? = null) : Style
+
+        /**
+         * An alert style with a tinted background.
+         */
+        data object Alert : Style
+
+        /**
+         * An alert style with the default background color.
+         */
+        data object AlertSolid : Style
+
+        /**
+         * A success style with a tinted background.
+         */
+        data object Success : Style
+
+        /**
+         * A success style with the default background color.
+         */
+        data object SuccessSolid : Style
+    }
+
+    /**
+     * Display a [BigIcon].
+     *
+     * @param style the style of the icon
+     * @param modifier the modifier to apply to this layout
+     */
+    @Composable
+    operator fun invoke(
+        style: Style,
+        modifier: Modifier = Modifier,
+    ) {
+        val backgroundColor = when (style) {
+            is Style.Default -> ElementTheme.colors.bigIconDefaultBackgroundColor
+            Style.AlertSolid, Style.SuccessSolid -> ElementTheme.colors.bgCanvasDefault
```

Comment:
I am wondering if using a transparent color should be preferable in this case. I let you decide.

Reply:
Transparent is probably better, although it kind of contradicts its name. I'm asking in the Compound room to clarify this.

Reply:
I just confirmed it with them, it's the other way around: the solid ones should have a solid background (duh), the others should be transparent. I'll change those.